source/usr/lib/mcvirt/test/auth_tests.py

```diff
@@ -219,3 +219,45 @@ def test_delete_non_existant_superuser(self):
             self.parser.parse_arguments('permission --delete-superuser %s --global' %
                                         self.test_user,
                                         mcvirt_instance=self.mcvirt)
+
+    def test_change_password(self):
+        """Change the password of a user through the parser"""
+        new_password = 'new-password-here'
+        self.parser.parse_arguments('user change-password --target-user %s --new-password %s' %
+            (self.TEST_USERNAME, new_password)
+        )
+
+        # Try to create an RPC connection with the old credentials and check an AuthenticationError
+        # is raised
+        with self.assertRaises(AuthenticationError):
+            new_rpc = Connection(self.TEST_USERNAME, self.TEST_PASSWORD)
+
+        # Create a new RPC connection with the new credentials
+        try:
+            new_rpc = Connection(self.TEST_USERNAME, new_password)
+        except AuthenticationError:
+            self.fail('Password not changed succesfully')
+
+    def test_add_new_user(self):
+        """Create a new user through the parser"""
+        username = 'brand-new-user'
+        password = 'pass'
+        self.parser.parse_arguments('user create %s --user-password %s' %
+            (username, password)
+        )
+
+        # Check the new user exists
+        self.assertTrue(username in MCVirtConfig().get_config()['users'])
+
+        # Delete the new user
+        new_user = self.user_factory.get_user_by_username(username)
+        self.rpc.annotate_object(new_user)
+        new_user.delete()
+
+    def test_remove_user_account(self):
+        """Delete a user through the parser"""
+        self.user_to_delete = self.create_test_user('user-to-delete', 'pass')
```

Comment:
I would put 'user-to-delete' in a variable :) Or possibly put a define a second test username in setUp and use it for 'user-to-delete' and 'brand-new-user'.

Then in the tearDown/setup, delete this user if it exists :)